src/main/java/nextstep/subway/line/LineService.java

```diff
@@ -90,23 +103,28 @@ private LineResponse createLineResponse(Line line) {
                 .id(line.getId())
                 .name(line.getName())
                 .color(line.getColor())
-                .stations(createStationResponses(line.getStationIds()))
+                .stations(createStationResponsesByIds(line.getStationIds()))
                 .build();
     }
 
-    private List<StationResponse> createStationResponses(List<Long> stationIds) {
-        return stationIds.stream()
-                .map(this::createStation)
+    private List<StationResponse> createStationResponsesByIds(List<Long> stationIds) {
+        List<Station> stations = stationIds.stream()
+                .map(this::lookUpStationBy)
                 .collect(Collectors.toList());
+        return createStationResponses(stations);
     }
 
-    private StationResponse createStation(Long stationId) {
-        Station station = stationRepository.findById(stationId)
-                .orElseThrow(NotExistStationException::new);
-        return new StationResponse(stationId, station.getName());
+    private List<StationResponse> createStationResponses(List<Station> stations) {
+        return stations.stream()
+                .map(LineService::createStationResponse)
```

Comment:
이와 같은 경우가 정적 팩토리 메서드 패턴을 쓰기 적절한 것 같네요!
```suggestion
                .map(StationResponse::from)
```